code/script.js

```diff
@@ -218,74 +257,196 @@ const generateBoard = () => {
+
+  //winOrLose.innerText += ` `;
   // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
+  // charactersInPlay = CHARACTERS;
   // What else should happen when we start the game?
 }
-
-// setting the currentQuestion object when you select something in the dropdown
+// Ylva - setting the currentQuestion object when you select something in the dropdown - "Does the person have..."
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+  // The category variable is used to store what option group (category) the question belongs to.
+  const category = questions.options[questions.selectedIndex].parentNode.label;
 
-  // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
+  // Ylva - this is a variable that stores the actual value of the question we've selected.
+  const value = questions.value;
+  // this above could also be written as:
+  // const value = questions.options[questions.selectedIndex].value;
 
+  // Ylva - This below is variables that stores the actual value of the current questions we've selected.
   currentQuestion = {
-    category: category,
-    // value: value
-  }
-}
+  category: category,
+  value: value,
+  };
+
+ // Ylva - to check the values of the currentQuestion object
+console.log(currentQuestion);
+};
+
+const playerNameSave = () => {
+playerName = document.getElementById("nameInput").value;
+console.log(playerName);
+};
 
 // This function should be invoked when you click on 'Find Out' button.
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
-
+  const { category, value } = currentQuestion;
+//if-statement limits the number of questions allowed to 3 questions in total
+  if (numberOfQuestions > 0) {
+    console.log(numberOfQuestions);
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
+  // See if we should keep or remove people based on that (true or false)
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
-
+  if (category === "Hair" || category === "Eyes") {
+   if (secret[category] === value) {
+    filterCharacters(true); //keep everyone with true according to hair/eye color
+   } else {
+   filterCharacters(); //remove everyone with false according to hair/eye color
+   }
+  } else if (category === "Accessories" || category === "Other") {
+if (secret[category].includes(value)) {
+filterCharacters(true); //keep everyone with true according to accessories/other
+} else {
+  filterCharacters(); //remove everyone with false according to accessories/other
```

Comment:
I don't think filterCharacters should be empty, it should be filterCharacters(false)

Reply:
@edv-rd I think you're right!
Right now there's so many things that doesn't work so I'm kind of rebuilding the whole thing...

As there are faults "everywhere" I find it hard to correct and all the code together just confuses me totally... 
As for now I'm trying to think like "okey - the timer is kind of the only thing working" so I'll look at some others students code (that works) and then rebuilding the whole game. 
And then I might (MAYBE) add the timer just for fun..